File: php/nodes/list_urls.php
<?php

function list_urls(array $edges): callable {
    return function ($state) use ($edges): Next {
        echo "\n--- URLs ---\n";
        foreach ($state['urls'] as $url) {
            echo "$url\n";
        }
        echo "------------\n";
        return new Next($edges['on_success'], $state);
    };
}

File: ruby/nodes/list_urls.rb
# frozen_string_literal: true

require_relative '../lib/ambler'

module ListUrls
  def self.node(edges)
    ->(state) {
      puts "\n--- URLs ---"
      state[:urls].each { |url| puts url }
      puts '------------'
      Ambler::Next.new(edges[:on_success], state)
    }
  end
end
